### PR TITLE
TPU hangs when saving optimizer/scheduler

### PR DESCRIPTION
**Pull request by LysandreJik**
Similarly to when saving a model state dict, the optimizer and scheduler should be saved using `xm.save` and behind an `xm.rendezvous`.

Additional fix: `pl.ParallelLoader` is not a `torch.utils.data.DataLoader`, and, therefore, must be reinitialized at each epoch.

### CODE REVIEW

**Review by jysohn23 — APPROVED** [review]
Thanks @LysandreJik !

**Review by julien-c — APPROVED** [review]

**Comment by julien-c** [comment]
Solid work, as discussed 👍

**Comment by ViktorAlm** [comment]
Nice! I tried two days ago and just skipped the checkpoints but something with the eval seemed to be messing things up(ForMultipleChoice) as well. WandB created 8 runs. dont know if its useful, im still figuring out alot of the stuff.
